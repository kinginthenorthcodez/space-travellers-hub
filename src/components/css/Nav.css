.nav-links {
  display: flex;
  gap: 2vw;
  align-items: center;
  color: blue;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  font-size: 2vw;
}

.nav-links:nth-child(4) {
  border-left: 2px solid grey;
}

.nav-links > a {
  text-decoration: none;
}

.active-nav {
  color: #00d7c6;
}
