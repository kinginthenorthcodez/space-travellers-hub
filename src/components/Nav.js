import { NavLink } from 'react-router-dom';
import './css/Nav.css';

const Nav = () => {
  const navs = [
    { id: 1, path: '/', text: 'Rockets' },
    { id: 2, path: '/dragons', text: 'Dragons' },
    { id: 3, path: 'missions', text: 'Missions' },
    { id: 4, path: '/myprofile', text: 'My Profile' },
  ];
  return (
    <div className="nav-links">
      {navs.map((nav) => (
        <NavLink
          className={(navlink) => (navlink.isActive ? 'active-nav' : '')}
          to={nav.path}
          key={nav.id}
        >
          {nav.text}
        </NavLink>
      ))}
    </div>
  );
};

export default Nav;
