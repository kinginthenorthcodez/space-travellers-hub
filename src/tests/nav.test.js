import React from 'react';
import { BrowserRouter as Rounter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Nav from '../components/Nav';

// Nav component tests
describe('Nav snapshot', () => {
  afterEach(cleanup);

  it('should render Nav correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Rounter>
          <Nav />
        </Rounter>
      </Provider>,
    );
    expect(asFragment(<Nav />)).toMatchSnapshot();
  });
});
