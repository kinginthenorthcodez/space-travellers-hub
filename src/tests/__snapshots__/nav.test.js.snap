// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav snapshot should render Nav correctly 1`] = `
<DocumentFragment>
  <div
    class="nav-links"
  >
    <a
      aria-current="page"
      class="active-nav"
      href="/"
    >
      Rockets
    </a>
    <a
      class=""
      href="/dragons"
    >
      Dragons
    </a>
    <a
      class=""
      href="/missions"
    >
      Missions
    </a>
    <a
      class=""
      href="/myprofile"
    >
      My Profile
    </a>
  </div>
</DocumentFragment>
`;
